osu.Game/Overlays/Chat/ChannelList/ChannelListSelector.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Overlays.Chat.ChannelList
+{
+    public class ChannelListSelector : OsuClickableContainer
+    {
+        public readonly BindableBool SelectorActive = new BindableBool();
+
+        private Box hoverBox = null!;
+        private Box selectBox = null!;
+
+        [Resolved]
+        private Bindable<ChannelSelectorState> selectorState { get; set; } = null!;
```

Comment:
I believe resolving bindables just one level down was frowned upon. Doing the `SelectorState = { BindTarget = SelectorState }` pattern in the pattern is preferred.

Reply:
Given this is also resolved for every `ChannelItem` I thought this would have been preferred instead setting bind target for each child element on construction.

Reply:
I mean the alternative is specifying `BindTarget` in something like two-three places? Not sure this is much of a gain at all, it's declarative code anyway.

I can't find examples from the past but passing down/using `BindTarget` was definitely preferred in a direct parent-child relationship like this is..

Reply:
Makes sense, I will change it.
Given `BindTarget` is preferred in this case, for future reference is there guidance on when using DI the better scenario for bindables so I don't make this mistake in the future?

Reply:
It's a bit fuzzy but I think for bindables the best practice is to use `BindTarget` as much as possible before it becomes unfeasible. Like if you have to ship that bind target 5 levels down, then yeah, DI away, but for simple parent-child glue code, even across three-four consecutive layers of parent-child, as long as you don't have to go out of your way to couple everything up, `BindTarget` is preferred.

Reply:
Thanks, that makes sense.
In that case, would it be worth changing the other chat components I have made that DI `Bindable<Channel>` to use `BindTarget` instead in a future PR since it is used in a similar way?

Reply:
Maybe? Not sure. You'd have to try and see how you go, I can see that falling into "maybe too tedious to not DI" category.

Reply:
I think the best way of figuring out which to use is to ask whether the two components (cacher and resolver) are directly related or not.

Examples of cases I'd say they aren't directly related:
- Caching at game, resolving from a screen (the screen is not created directly by the game but another screen)
- Caching at `DrawableRuleset`, resolving from a `DrawableHitObject` (hit objects are created by a means which `DrawableRuleset` doesn't have direct control over)

Another edge case I'd usually say DI makes more sense is when the whole class is to be exposed (ie. caching of `SongSelect` for use by one or more children). In these cases I'd probably be okay with DI over passing in `this`.